staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -152,27 +150,14 @@ func (f *FieldManager) Apply(liveObj runtime.Object, patch []byte, fieldManager
 	if err != nil {
 		return nil, fmt.Errorf("failed to decode managed fields: %v", err)
 	}
-	// Check that the patch object has the same version as the live object
-	patchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
-
-	if err := yaml.Unmarshal(patch, &patchObj.Object); err != nil {
-		return nil, fmt.Errorf("error decoding YAML: %v", err)
-	}
-	if patchObj.GetAPIVersion() != f.groupVersion.String() {
-		return nil,
-			errors.NewBadRequest(
-				fmt.Sprintf("Incorrect version specified in apply patch. "+
-					"Specified patch version: %s, expected: %s",
-					patchObj.GetAPIVersion(), f.groupVersion.String()))
-	}
 
 	liveObjVersioned, err := f.toVersioned(liveObj)
 	if err != nil {
 		return nil, fmt.Errorf("failed to convert live object to proper version: %v", err)
 	}
 	internal.RemoveObjectManagedFields(liveObjVersioned)
 
-	patchObjTyped, err := f.typeConverter.YAMLToTyped(patch)
+	patchObjTyped, err := f.typeConverter.ObjectToTyped(patchObj)
```

Comment:
@jennybuckley is this going to work or will we have problems with orders?

Reply:
could the failing test be related to this @apelisse ?

Reply:
I think we need to create the TypedValue directly from YAML to preserve float/int distinction, since converting to untyped first (which is what patchObj is) is going to convert all the ints to floats

Reply:
In the failure ```Failed to re-apply object using Apply patch: Apply failed with 1 conflict: conflict with "apply_test": .spec.ports[port=8080,protocol="UDP"].port```

Reply:
is caused by it thinking the value of port changed because it's changing from an int to a float

Reply:
Exactly, that's the problem you're running into:
```
apply_test.go:133: Failed to re-apply object using Apply patch: Apply failed with 1 conflict: conflict with "apply_test": .spec.ports[port=8080,protocol="UDP"].port
```

The thing is confused because port is float and/or int, and it gets confused ...

Reply:
Looks like I lost the race :disappointed:

Reply:
OK I think you have two options here:
1. You could move everything back into `FieldManager` the way it was before, but then it's really inconsistent with the other call-sites,
2. You could keep this code, but pass-in the patchBytes to the fieldManager, and re-deserialize again there.

While 2. requires deserializing twice, I think it makes more sense for applyPatcher to be more consistent with the rest of the patchers, so I would take that option.